### PR TITLE
ascanrulesBeta: add example alerts to HttpOnlySiteScanRule

### PR DESCRIPTION
**Pull request by GRVial**
## Overview
Added example alerts for [HttpOnlySiteScanRule](https://www.zaproxy.org/docs/alerts/10106/).

## Related Issues
[#6119](https://github.com/zaproxy/zaproxy/issues/6119)

## Checklist
- [ ] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/HttpOnlySiteScanRule.java [hunk 1]
```diff
@@ -221,4 +225,17 @@ private static int getPort(HttpMessage message) {
         }
         return port;
     }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        String exampleUri = "http://example.com";
+        return List.of(
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/HttpOnlySiteScanRule.java:232]: Should return a single alert.

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - The following scan rules were added, having been promoted from Alpha:
   - Server Side Request Forgery
   - Text4shell (CVE-2022-42889)
+- HttpOnlySiteScanRule now includes example alert functionality for documentation generation purposes (Issue 6119).
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/CHANGELOG.md:12]: Should be merged with the existing note on the same topic

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/HttpOnlySiteScanRule.java [hunk 3]
```diff
@@ -108,18 +109,21 @@ public Map<String, String> getAlertTags() {
         return ALERT_TAGS;
     }
 
-    public void raiseAlert(HttpMessage newRequest, String message) {
-        String newUri = newRequest.getRequestHeader().getURI().toString();
+    private AlertBuilder createAlert(HttpMessage newRequest, String message, String newUri) {
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/HttpOnlySiteScanRule.java:112]: If you’re passing the whole message why pass the uri?

Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/HttpOnlySiteScanRule.java:112]: …or vice versa. Only one should be needed.

**Comment by thc202** [comment]
This needs to be rebased.

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## [47] - 2023-07-20
 ### Added
-- The Source Code Disclosure - File Inclusion now includes example alert functionality for documentation generation purposes (Issue 6119).
+- The Source Code Disclosure - File Inclusion and HttpOnlySiteScanRule now includes example alert functionality for documentation generation purposes (Issue 6119).
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/CHANGELOG.md:11]:
```suggestion
- The Source Code Disclosure - File Inclusion and HTTP Only Site rules now include example alert functionality for documentation generation purposes (Issue 6119).
```

Comment (thc202) [reply addOns/ascanrulesBeta/CHANGELOG.md]: It was not the final push, this needs to be moved to Unreleased.

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**